kubernetes/iceberg/dns-solution.md

```diff
@@ -0,0 +1,60 @@
+# Kubernetes DNS Solution for Iceberg + MinIO Integration
+
+This document explains the solution implemented to handle DNS resolution challenges in the Kubernetes deployment of the Redpanda Iceberg lab.
+
+## The problem
+
+The Iceberg REST catalog client uses bucket-style S3 URLs when accessing MinIO, generating hostnames like:
+`redpanda.iceberg-minio-hl.iceberg-lab.svc.cluster.local`
+
+These aren't valid Kubernetes DNS names because they contain dots in the subdomain part, causing `UnknownHostException` errors.
+
+## The solution
+
+### 1. Static hostAliases
+
+```yaml
+hostAliases:
+  - ip: "10.244.4.3"  # MinIO pod IP
+    hostnames:
+    - "redpanda.iceberg-minio-hl.iceberg-lab.svc.cluster.local"
+```
+
+**Problem**: Pod IPs are ephemeral and change when pods restart.
+
+### 2. Dynamic script
+
+The `deploy-iceberg-rest-catalog.sh` and `deploy-spark.sh` scripts provide:
+
+- **Multi-strategy IP resolution**: Service ClusterIP → Pod IP → Endpoints
+- **Connection validation**: Tests MinIO and REST catalog connectivity
+- **Error handling**: Comprehensive validation and clear error messages
+- **Health checks**: Kubernetes probes for better reliability
+
+## Production considerations
+
+Consider additional approaches:
```

Comment:
This feels weird to include in a lab. Is it standard practice?

Reply:
Removed